File: trip-backend/src/models/feedback.py
"""Feedback model (feedbacks table)"""

from sqlalchemy import Column, Integer, String, JSON, ForeignKey, Text, Index, UniqueConstraint
from sqlalchemy.orm import relationship

from src.models.base import Base, BaseModel


class Feedback(Base, BaseModel):
    """Feedback model matching Prisma schema"""
    
    __tablename__ = "feedbacks"
    __table_args__ = (
        UniqueConstraint("user_id", "message_id", name="uq_feedback_user_message"),
        Index("idx_feedback_message", "message_id"),
        Index("idx_feedback_rating_created", "rating", "created_at"),
        Index("idx_feedback_user_created", "user_id", "created_at"),
        {"comment": "反馈表"},
    )
    
    # Fields
    user_id = Column(
        "user_id",
        Integer,
        ForeignKey("users.id"),
        nullable=False,
        comment="用户ID（外键）"
    )
    message_id = Column(
        "message_id",
        Integer,
        ForeignKey("messages.id", ondelete="CASCADE"),
        nullable=False,
        comment="消息ID（外键）"
    )
    conversation_id = Column(
        "conversation_id",
        Integer,
        nullable=False,
        comment="对话ID"
    )
    rating = Column(
        Integer,
        nullable=False,
        comment="评分（1=点赞，-1=点踩）"
    )
    comment = Column(
        String(500),
        nullable=True,
        comment="评论"
    )
    tags = Column(
        JSON,
        nullable=True,
        comment="标签（JSON数组）"
    )
    
    # Relationships
    user = relationship("User", back_populates="feedbacks")
    message = relationship("Message", back_populates="feedbacks")
    
    def __repr__(self):
        return f"<Feedback(id={self.id}, rating={self.rating})>"

File: trip-backend-java/src/main/java/com/trip/backend/domain/entity/Feedback.java
package com.trip.backend.domain.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.OffsetDateTime;
import java.util.List;

/**
 * 反馈实体（对应 Python models/feedback.py）
 * - tags 为 JSONB 列
 * - 唯一约束 (user_id, message_id)
 * - ondelete CASCADE（message_id FK）
 */
@Entity
@Table(name = "feedbacks", indexes = {
    @Index(name = "idx_feedbacks_message_id", columnList = "message_id"),
    @Index(name = "idx_feedbacks_rating_created", columnList = "rating, created_at"),
    @Index(name = "idx_feedbacks_user_created", columnList = "user_id, created_at")
})
@Getter
@Setter
public class Feedback {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long userId;

    @Column(nullable = false)
    private Long messageId;

    @Column(nullable = false)
    private Long conversationId;

    @Column(nullable = false)
    private Integer rating; // 1 或 -1

    @Column(length = 500)
    private String comment;

    // JSONB 列
    @JdbcTypeCode(SqlTypes.JSON)
    @Column(columnDefinition = "jsonb")
    private List<String> tags;

    @Column(nullable = false)
    private OffsetDateTime createdAt;

    protected Feedback() {}

    @PrePersist
    protected void onCreate() {
        this.createdAt = OffsetDateTime.now();
    }
}
